packages/app/src/common/hooks/use-setup-tx.ts

```diff
@@ -0,0 +1,165 @@
+    },
+    []
+  );
+
+  /**
+   * Apps can specify a `stxAddress` in a transaction request.
+   * If the user has a matching account, use that account by default.
+   */
+  const handleAccountSwitch = useRecoilCallback(
+    ({ snapshot, set }) => async () => {
+      const payload = await snapshot.getPromise(transactionPayloadStore);
+      if (!payload?.stxAddress || !payload.network) return;
+      const wallet = await snapshot.getPromise(walletStore);
+      if (!wallet) return;
+      const transactionVersion = payload.network.version;
+      let foundIndex: number | undefined = undefined;
+      wallet.accounts.forEach((account, index) => {
+        const address = getStxAddress({ account, transactionVersion });
+        if (address === payload.stxAddress) {
+          foundIndex = index;
+        }
+      });
+      if (foundIndex !== undefined) {
+        console.debug('switching to index', foundIndex);
+        set(currentAccountIndexStore, foundIndex);
+      } else {
+        console.warn(
+          'No account matches the STX address provided in transaction request:',
+          payload.stxAddress
+        );
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    void handleNetworkSwitch();
+    void handleAccountSwitch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [requestToken]);
+
+  const handlePostConditions = useRecoilCallback(
+    ({ snapshot, set }) => async (currentAddress?: string) => {
+      const payload = await snapshot.getPromise(transactionPayloadStore);
+      if (!payload) return;
+      if (!currentAddress) return;
+      const existingPostConditions = await snapshot.getPromise(postConditionsStore);
+      // change existing post conditions to current account
+      if (existingPostConditions.length) {
+        console.debug('Swapping post conditions address');
+        const newConditions: PostCondition[] = existingPostConditions.map(postCondition => {
+          return {
+            ...postCondition,
+            principal: parsePrincipalString(currentAddress),
+          };
+        });
+        set(postConditionsStore, newConditions);
+        return;
+      } else if (payload.postConditions?.length) {
+        console.debug('Setting up post conditions for transaction request');
+        const newConditions: PostCondition[] = payload.postConditions.map(postCondition => {
+          let newCondition: PostCondition;
+          if (typeof postCondition === 'string') {
+            const reader = BufferReader.fromBuffer(Buffer.from(postCondition, 'hex'));
+            newCondition = deserializePostCondition(reader);
+          } else {
+            newCondition = { ...postCondition };
+          }
+          // override principal with current user
+          newCondition = {
+            ...newCondition,
+            principal: parsePrincipalString(currentAddress),
+          };
+          return newCondition;
+        });
+        set(postConditionsStore, newConditions);
+      }
```

Comment:
Nit pick, but this is kinda mixing programming styles imo. Mapping an array, functional, then reassigning a ref, imperative. No action needed, just an observation that it'd look cleaner with a const lol.